libs/wingsdk/src/target-sim/bucket.inflight.ts

```diff
@@ -269,6 +269,17 @@ export class Bucket implements IBucketClient, ISimulatorResourceInstance {
     });
   }
 
+  public async rename(srcKey: string, dstKey: string): Promise<void> {
+    if (srcKey === dstKey) {
+      throw new Error(
+        `Renaming an object to its current name is not a valid operation (srcKey=${srcKey}, dstKey=${dstKey}).`
+      );
+    }
+
+    await this.copy(srcKey, dstKey);
+    await this.delete(srcKey);
+  }
```

Comment:
is there a way to avoid this code being duplicated for each of the targets?

Reply:
How would you do that?

Reply:
maybe implementing a base class?